test/unit/event/src/EventTest.cpp

```diff
@@ -227,44 +236,260 @@ TEMPLATE_LIST_TEST_CASE("waitForEventThatAlreadyFinishedShouldBeSkipped", "[even
+            REQUIRE(!alpaka::isComplete(k1_0));
+
+            alpaka::wait(q2, e1);
+            alpaka::enqueue(q2, e2);
+            // q2 = [->e1,e2]
+            alpaka::enqueue(q1, k1_1);
+            alpaka::enqueue(q1, e1);
+            // q1 = [k1_0,e1,k1_1,e1_new]
+            alpaka::enqueue(q3, k3);
+            alpaka::wait(q3, e1);
+            alpaka::enqueue(q3, e3);
+
+            // q1 = [k1_0,e1,k1_1,e1_new]
+            // q2 = [k2,->e1,e2]
+            // q3 = [k3,->e1_new,e3]
+            REQUIRE(!alpaka::isComplete(k1_0));
+            REQUIRE(!alpaka::isComplete(k1_1));
+            REQUIRE(!alpaka::isComplete(k2));
+            REQUIRE(!alpaka::isComplete(k3));
+            REQUIRE(!alpaka::isComplete(e1));
+            REQUIRE(!alpaka::isComplete(e2));
+            REQUIRE(!alpaka::isComplete(e3));
+
+            k3.trigger();
+
+            // q1 = [k1_0,e1,k1_1,e1_new]
+            // q2 = [k2,->e1,e2]
+            // q3 = [->e1_new,e3]
+            REQUIRE(!alpaka::isComplete(k1_0));
+            REQUIRE(!alpaka::isComplete(k1_1));
+            REQUIRE(!alpaka::isComplete(k2));
+            REQUIRE(alpaka::isComplete(k3));
+            REQUIRE(!alpaka::isComplete(e1));
+            REQUIRE(!alpaka::isComplete(e2));
+            REQUIRE(!alpaka::isComplete(e3));
+
+            k2.trigger();
+
+            // q1 = [k1_0,e1,k1_1,e1_new]
+            // q2 = [->e1,e2]
+            // q3 = [->e1_new,e3]
+            REQUIRE(!alpaka::isComplete(k1_0));
+            REQUIRE(!alpaka::isComplete(k1_1));
+            REQUIRE(alpaka::isComplete(k2));
+            REQUIRE(alpaka::isComplete(k3));
+            REQUIRE(!alpaka::isComplete(e1));
+            REQUIRE(!alpaka::isComplete(e2));
+            REQUIRE(!alpaka::isComplete(e3));
+
+            // After the kernel k1_0 is released e3 is not allowed to be ready because q3 depends on the oldest e1_new
+            // state.
+            k1_0.trigger();
+            // q1 = [k1_1,e1_new]
+            // q2 = []
+            // q3 = [->e1_new,e3]
+
+            REQUIRE(alpaka::isComplete(k1_0));
+            REQUIRE(!alpaka::isComplete(k1_1));
+            REQUIRE(alpaka::isComplete(k2));
+            REQUIRE(alpaka::isComplete(k3));
+            REQUIRE(!alpaka::isComplete(e1));
+            REQUIRE(alpaka::isComplete(e2));
+            REQUIRE(!alpaka::isComplete(e3));
+
+            k1_1.trigger();
+            // q1 = []
             // q2 = []
+            REQUIRE(alpaka::isComplete(k1_0));
+            REQUIRE(alpaka::isComplete(k1_1));
+            REQUIRE(alpaka::isComplete(k2));
+            REQUIRE(alpaka::isComplete(k3));
+            REQUIRE(alpaka::isComplete(e1));
+            REQUIRE(alpaka::isComplete(e2));
+            REQUIRE(alpaka::isComplete(e3));
+        }
+        else
+        {
+            std::cerr << "Can not execute test because CU_DEVICE_ATTRIBUTE_CAN_USE_STREAM_MEM_OPS is not supported!"
+                      << std::endl;
```

Comment:
```suggestion
            std::cerr << "Cannot execute test because CU_DEVICE_ATTRIBUTE_CAN_USE_STREAM_MEM_OPS is not supported by the device!"
                      << std::endl;
```